src/customnodesession.hpp

```diff
@@ -53,6 +54,6 @@ class CustomNodeSession : public NodeSession {
 
 private:
     static void releaseTensorResources(const struct CustomNodeTensor* tensor, const NodeLibrary& library, void* customNodeLibraryInternalManager);
-    Status createBlob(const struct CustomNodeTensor* tensor, std::shared_ptr<ov::runtime::Tensor>& resultBlob, const NodeLibrary& library, void* customNodeLibraryInternalManager);
+    Status createBlob(const struct CustomNodeTensor* tensor, std::shared_ptr<ov::runtime::Tensor>& resultBlob, const NodeLibrary& library, std::shared_ptr<CNLIMWrapper>& customNodeLibraryInternalManager);
```

Comment:
I see alot of places use shared pointer reference. Do we not want to track usage (use counter) of such places? If so, why?

Reply:
I suppose that we could pass void* instead of reference to shared_ptr in some places.

Reply:
I mean, are we sure that we only need to keep one copy of shared_ptr in CustomNode object and pass reference in all other places?

Reply:
Pipeline object with CustomNode objects are destroyed after its creation and execution. Since all functions are part of that execution I think that there is no case in which that shared_ptr do not exist. Correct me if I am wrong.

Reply:
Agree